SampleApplications/2023/ServerlessDocumentAnalysis/functions/src/ProcessTextractExpenseResults/Function.cs

```diff
@@ -0,0 +1,86 @@
+using Amazon.Lambda.Annotations;
+using Amazon.Lambda.Core;
+using Amazon.XRay.Recorder.Handlers.AwsSdk;
+using AWS.Lambda.Powertools.Logging;
+using AWS.Lambda.Powertools.Metrics;
+using AWS.Lambda.Powertools.Tracing;
+using DocProcessing.Shared.Model.Data.Expense;
+using System.Threading.Tasks;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]
+[assembly: LambdaGlobalProperties(GenerateMain = true)]
+
+namespace ProcessTextractExpenseResults;
+
+public class Function(ITextractService textractService, IDataService dataService)
+{
+
+    private readonly ITextractService _textractService = textractService;
+    private readonly IDataService _dataService = dataService;
+
+    static Function()
+    {
+        AWSSDKHandler.RegisterXRayForAllServices();
+    }
+
+    [Tracing]
+    [Metrics]
+    [Logging]
+    [LambdaFunction]
+    public async Task<IdMessage> FunctionHandler(IdMessage input, ILambdaContext _context)
+    {
+        var processData = await _dataService.GetData<ProcessData>(input.Id).ConfigureAwait(false);
+
+        // Get the step functions Result
+        var textractExpenseModel = await _textractService.GetExpenses(processData.OutputBucket, processData.ExpenseOutputKey).ConfigureAwait(false);
+
+        // Get Expense Data and save to the DB
+        foreach (var id in textractExpenseModel.GetExpenseReportIndexes())
+        {
+            var report = new DocumentExpenseReport();
+
+            //Add Scalar Values            
+            foreach (var scalarSummaryField in textractExpenseModel.GetScalarSummaryFields(id))
```

Comment:
Consider extracting to method instead/in-addition to the "header" comments

Reply:
Refactored this significantly.